conformant = "Partial"
notes = """
Incorrectly reports error for type alias defined with ParamSpec.
Incorrectly rejects some valid type aliases when used in annotations.
Incorrectly evaluates generic type alias with ParamSpec with missing type argument.
Does not report invalid specialization of generic type aliases.
Does not report error for attempt to instantiate union type alias.
Does not report invalid specialization of already-specialized generic type alias.
"""
output = """
aliases_implicit.py:39:26 Incompatible parameter type [6]: In call `typing.GenericMeta.__getitem__`, for 1st positional argument, expected `Tuple[typing.Any, Type[Variable[$synthetic_attribute_resolution_variable]]]` but got `Tuple[object, TypeVar]`.
aliases_implicit.py:43:27 Incompatible parameter type [6]: In call `typing.GenericMeta.__getitem__`, for 1st positional argument, expected `Tuple[typing.Any, Type[Variable[$synthetic_attribute_resolution_variable]]]` but got `Tuple[ParamSpec, None]`.
aliases_implicit.py:55:8 Undefined or invalid type [11]: Annotation `GoodTypeAlias9` is not defined as a type.
aliases_implicit.py:59:9 Undefined or invalid type [11]: Annotation `GoodTypeAlias13` is not defined as a type.
aliases_implicit.py:63:25 Incompatible parameter type [6]: In call `typing.GenericMeta.__getitem__`, for 1st positional argument, expected `Type[Variable[_T]]` but got `UnionType`.
aliases_implicit.py:65:26 Incompatible parameter type [6]: In call `typing.GenericMeta.__getitem__`, for 1st positional argument, expected `Type[Variable[_T_co](covariant)]` but got `Tuple[Type[str], typing.Any]`.
aliases_implicit.py:66:26 Incompatible parameter type [6]: In call `typing.GenericMeta.__getitem__`, for 1st positional argument, expected `Type[Variable[_T_co](covariant)]` but got `Tuple[Type[int], Type[int], Type[int], Type[str]]`.
aliases_implicit.py:69:29 Incompatible parameter type [6]: In call `typing.GenericMeta.__getitem__`, for 1st positional argument, expected `Tuple[typing.Any, Type[Variable[$synthetic_attribute_resolution_variable]]]` but got `Tuple[List[Type[Union[int, str]]], None]`.
aliases_implicit.py:73:30 Incompatible parameter type [6]: In call `typing.GenericMeta.__getitem__`, for 1st positional argument, expected `Tuple[typing.Any, Type[Variable[$synthetic_attribute_resolution_variable]]]` but got `Tuple[typing.Any, None]`.
aliases_implicit.py:107:8 Undefined or invalid type [11]: Annotation `BadTypeAlias1` is not defined as a type.
aliases_implicit.py:108:8 Undefined or invalid type [11]: Annotation `BadTypeAlias2` is not defined as a type.
aliases_implicit.py:109:8 Undefined or invalid type [11]: Annotation `BadTypeAlias3` is not defined as a type.
aliases_implicit.py:110:8 Undefined or invalid type [11]: Annotation `BadTypeAlias4` is not defined as a type.
aliases_implicit.py:111:8 Undefined or invalid type [11]: Annotation `BadTypeAlias5` is not defined as a type.
aliases_implicit.py:112:8 Undefined or invalid type [11]: Annotation `BadTypeAlias6` is not defined as a type.
aliases_implicit.py:113:8 Undefined or invalid type [11]: Annotation `BadTypeAlias7` is not defined as a type.
aliases_implicit.py:114:8 Undefined or invalid type [11]: Annotation `BadTypeAlias8` is not defined as a type.
aliases_implicit.py:115:8 Undefined or invalid type [11]: Annotation `BadTypeAlias9` is not defined as a type.
aliases_implicit.py:116:9 Undefined or invalid type [11]: Annotation `BadTypeAlias10` is not defined as a type.
aliases_implicit.py:117:9 Undefined or invalid type [11]: Annotation `BadTypeAlias11` is not defined as a type.
aliases_implicit.py:118:9 Undefined or invalid type [11]: Annotation `BadTypeAlias12` is not defined as a type.
aliases_implicit.py:119:9 Undefined or invalid type [11]: Annotation `BadTypeAlias13` is not defined as a type.
aliases_implicit.py:120:9 Undefined or invalid type [11]: Annotation `BadTypeAlias14` is not defined as a type.
"""
conformance_automated = "Fail"
errors_diff = """
Line 77: Expected 1 errors
Line 78: Expected 1 errors
Line 79: Expected 1 errors
Line 80: Expected 1 errors
Line 81: Expected 1 errors
Line 82: Expected 1 errors
Line 134: Expected 1 errors
Line 136: Expected 1 errors
Line 39: Unexpected errors ['aliases_implicit.py:39:26 Incompatible parameter type [6]: In call `typing.GenericMeta.__getitem__`, for 1st positional argument, expected `Tuple[typing.Any, Type[Variable[$synthetic_attribute_resolution_variable]]]` but got `Tuple[object, TypeVar]`.']
Line 43: Unexpected errors ['aliases_implicit.py:43:27 Incompatible parameter type [6]: In call `typing.GenericMeta.__getitem__`, for 1st positional argument, expected `Tuple[typing.Any, Type[Variable[$synthetic_attribute_resolution_variable]]]` but got `Tuple[ParamSpec, None]`.']
Line 55: Unexpected errors ['aliases_implicit.py:55:8 Undefined or invalid type [11]: Annotation `GoodTypeAlias9` is not defined as a type.']
Line 59: Unexpected errors ['aliases_implicit.py:59:9 Undefined or invalid type [11]: Annotation `GoodTypeAlias13` is not defined as a type.']
Line 63: Unexpected errors ['aliases_implicit.py:63:25 Incompatible parameter type [6]: In call `typing.GenericMeta.__getitem__`, for 1st positional argument, expected `Type[Variable[_T]]` but got `UnionType`.']
Line 65: Unexpected errors ['aliases_implicit.py:65:26 Incompatible parameter type [6]: In call `typing.GenericMeta.__getitem__`, for 1st positional argument, expected `Type[Variable[_T_co](covariant)]` but got `Tuple[Type[str], typing.Any]`.']
Line 66: Unexpected errors ['aliases_implicit.py:66:26 Incompatible parameter type [6]: In call `typing.GenericMeta.__getitem__`, for 1st positional argument, expected `Type[Variable[_T_co](covariant)]` but got `Tuple[Type[int], Type[int], Type[int], Type[str]]`.']
Line 69: Unexpected errors ['aliases_implicit.py:69:29 Incompatible parameter type [6]: In call `typing.GenericMeta.__getitem__`, for 1st positional argument, expected `Tuple[typing.Any, Type[Variable[$synthetic_attribute_resolution_variable]]]` but got `Tuple[List[Type[Union[int, str]]], None]`.']
Line 73: Unexpected errors ['aliases_implicit.py:73:30 Incompatible parameter type [6]: In call `typing.GenericMeta.__getitem__`, for 1st positional argument, expected `Tuple[typing.Any, Type[Variable[$synthetic_attribute_resolution_variable]]]` but got `Tuple[typing.Any, None]`.']
"""
